pkg/telemetry/info.go

```diff
@@ -0,0 +1,135 @@
+package telemetry
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+// Info represents an info metric - a gauge that always has value 1.0
+// Info metrics are used to expose textual information as labels.
+type Info struct {
+	gauge *FloatGauge
+	attrs []attribute.KeyValue
+}
+
+// InfoConfig configures an info metric
+type InfoConfig struct {
+	Name        string
+	Description string
+	Labels      map[string]string
+}
+
+// NewInfo creates a new info metric that always reports 1.0
+// This is useful for exposing version info, addresses, and other metadata
+func NewInfo(meter metric.Meter, cfg InfoConfig) (*Info, error) {
+	gauge, err := NewFloatGauge(meter, FloatGaugeConfig{
+		Name:        cfg.Name,
+		Description: cfg.Description,
+		Unit:        "1", // Info metrics are dimensionless
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	attrs := make([]attribute.KeyValue, 0, len(cfg.Labels))
+	for k, v := range cfg.Labels {
+		attrs = append(attrs, attribute.String(k, v))
+	}
+
+	return &Info{
+		gauge: gauge,
+		attrs: attrs,
+	}, nil
+}
+
+// Record records the info metric with value 1.0
+func (i *Info) Record(ctx context.Context) {
+	i.gauge.Record(ctx, 1.0, i.attrs...)
+}
+
+// Update updates the info metric with new label values
+func (i *Info) Update(ctx context.Context, labels map[string]string) {
+	attrs := make([]attribute.KeyValue, 0, len(labels))
+	for k, v := range labels {
+		attrs = append(attrs, attribute.String(k, v))
+	}
+	i.attrs = attrs
+	i.Record(ctx)
+}
```

Comment:
since this is an `Info` metric, I think the most common way of using it will be updating the value of a single label. As it is, `Record` is kinda useless and `Update` requires passing all the labels. I think a better API would be having a single `Record` method that accepts a map of labels and then merges it with the one already present in the `Info`.

Reply:
I'll leave it open for now, but that's a good point to consider if we keep using it with the same labels again and again.